doc/tutorials/12-constant_pH/12-constant_pH.ipynb

```diff
@@ -120,20 +196,75 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "K = 1e-4\n",
-    "pK = -np.log10(K)"
+    "# create the particles\n",
+    "##################################################\n",
+    "# we need to define bonds before creating polymers\n",
+    "from espressomd.interactions import HarmonicBond\n",
+    "hb= HarmonicBond(k=30, r_0=1.0)\n",
+    "system.bonded_inter.add(hb)\n",
+    "\n",
+    "# create the polymer composed of ionizable acid groups, initially in the ionized state\n",
+    "from espressomd import polymer\n",
+    "polymers = polymer.positions(n_polymers=1,\n",
+    "                             beads_per_chain=N_acid,\n",
+    "                             bond_length=0.9, seed=23)\n",
+    "for p in polymers:\n",
+    "    for i, m in enumerate(p):\n",
+    "       id = len(system.part)\n",
+    "       system.part.add(id=id, pos=m, type=TYPE_A, q=q_A)\n",
+    "       if i > 0:\n",
+    "           system.part[id].add_bond((hb, id - 1))\n",
+    "\n",
+    "# add the corresponding number of H+ ions\n",
+    "for i in range(N_acid):\n",
+    "        system.part.add(pos=np.random.random(3)*Box_L_in_sigma, type=TYPE_B, q=q_B)\n",
+    "\n",
+    "# add salt ion pairs\n",
+    "for i in range(N_salt):\n",
+    "        system.part.add(pos=np.random.random(3)*Box_L_in_sigma, type=TYPE_Na, q=q_Na)\n",
+    "        system.part.add(pos=np.random.random(3)*Box_L_in_sigma, type=TYPE_Cl, q=q_Cl)\n",
+    "\n",
+    "# set up the WCA interaction between all particle pairs\n",
+    "if USE_WCA:\n",
+    "    types=[TYPE_HA, TYPE_A, TYPE_B, TYPE_Na, TYPE_Cl]\n",
+    "    for type_1 in types:\n",
+    "        for type_2 in types:\n",
+    "            system.non_bonded_inter[type_1, type_2].lennard_jones.set_params(\n",
+    "                epsilon=1.0, sigma=1.0,\n",
+    "                cutoff=2**(1.0 / 6), shift=\"auto\")\n",
+    "\n",
+    "# run a steepest descent minimization to relax overlaps\n",
+    "system.integrator.set_steepest_descent(\n",
+    "    f_max=0, gamma=0.1, max_displacement=0.1)\n",
+    "system.integrator.run(20)\n",
+    "system.integrator.set_vv()  # to switch back to velocity Verlet\n",
+    "\n",
+    "# run a short integration before using the electrostatics\n",
```

Comment:
the comment should explain why there are integrations before setting up electrostatics